NeuroID/src/advancedDeviceLib/java/com/neuroid/tracker/service/NIDAdvancedDeviceApiService.kt

```diff
@@ -0,0 +1,11 @@
+package com.neuroid.tracker.service
+
+import okhttp3.ResponseBody
+import retrofit2.Call
+import retrofit2.http.GET
+import retrofit2.http.Path
+
+interface NIDAdvancedDeviceApiService {
+    @GET("/a/{key}")
+    fun getADVKey(@Path("key") key: String): Call<ResponseBody>
```

Comment:
you can replace <ResponserBody> with a data model <AVDKeyData>. The ADVKeyData can then be a data class with annotation indicating type and name of the keys in the model from the returning JSON. 

`data class ADVKeyNetworkResponse(@field: Json(name="status")var status: String, @field: Json(name="key")var key: String)`

This the execute will deserialize the response to a ADVKeyNetworkResponse and you can dump the deserializing code in lines 64-79 in the NIDAdvancedDeviceNetworkService.retryRequests().

Reply:
Updated!